command/operator_snapshot_inspect.go

```diff
@@ -5,17 +5,57 @@ package command
 
 import (
 	"fmt"
+	"io"
 	"os"
+	"sort"
+	"strconv"
 	"strings"
+	"time"
 
+	"github.com/hashicorp/go-msgpack/codec"
 	"github.com/hashicorp/nomad/helper/snapshot"
+	"github.com/hashicorp/nomad/nomad"
+	"github.com/hashicorp/raft"
 	"github.com/posener/complete"
 )
 
 type OperatorSnapshotInspectCommand struct {
 	Meta
 }
 
+type typeStats struct {
+	Name  string
+	Sum   int
+	Count int
+}
+
+type SnapshotInspectFormat struct {
+	Meta  *raft.SnapshotMeta
+	Stats []typeStats
+}
+
+// SnapshotInfo is used for passing snapshot stat
+// information between functions
+type SnapshotInfo struct {
+	Stats     map[nomad.SnapshotType]typeStats
+	TotalSize int
+}
+
+// countingReader helps keep track of the bytes we have read
+// when reading snapshots
+type countingReader struct {
+	wrappedReader io.Reader
+	read          int
+}
+
+func (r *countingReader) Read(p []byte) (n int, err error) {
+	n, err = r.wrappedReader.Read(p)
+	if err == nil {
+		r.read += n
+	}
+	return n, err
+}
+
 func (c *OperatorSnapshotInspectCommand) Help() string {
```

Comment:
Seeing as the command now shows more information and this is sorted, I think it would be good to expand on the command help output.